src/textual/widgets/_data_table.py

```diff
@@ -650,33 +995,73 @@ def add_row(self, *cells: CellType, height: int = 1) -> None:
         if cell_now_available and visible_cursor:
             self._highlight_cursor()
 
-    def add_rows(self, rows: Iterable[Iterable[CellType]]) -> None:
-        """Add a number of rows.
+        self.check_idle()
+        return row_key
+
+    def add_columns(self, *labels: TextType) -> list[ColumnKey]:
+        """Add a number of columns.
+
+        Args:
+            *labels: Column headers.
+
+        Returns:
+            A list of the keys for the columns that were added. See
+                the `add_column` method docstring for more information on how
+                these keys are used.
+        """
+        column_keys = []
+        for label in labels:
+            column_key = self.add_column(label, width=None)
+            column_keys.append(column_key)
+        return column_keys
+
+    def add_rows(self, rows: Iterable[Iterable[CellType]]) -> list[RowKey]:
+        """Add a number of rows at the bottom of the DataTable.
 
         Args:
             rows: Iterable of rows. A row is an iterable of cells.
+
+        Returns:
+            A list of the keys for the rows that were added. See
+                the `add_row` method docstring for more information on how
+                these keys are used.
         """
+        row_keys = []
         for row in rows:
-            self.add_row(*row)
+            row_key = self.add_row(*row)
+            row_keys.append(row_key)
+        return row_keys
 
     def on_idle(self) -> None:
+        """Runs when the message pump is empty. We use this for
+        some expensive calculations like re-computing dimensions of the
+        whole DataTable and re-computing column widths after some cells
+        have been updated. This is more efficient in the case of high
+        frequency updates, ensuring we only do expensive computations once."""
```

Comment:
May want reformatting to confirm with [our docstring guidelines](https://github.com/Textualize/textual/wiki/Documentation-standards#class).